src/OrchardCore.Modules/OrchardCore.Demo/Startup.cs

```diff
@@ -102,6 +102,12 @@ public override void ConfigureServices(IServiceCollection services)
                 // Add a custom folder route
                 options.Conventions.AddAreaFolderRoute("OrchardCore.Demo", "/", "Demo");
 
+                 // Add a custom folder route
+                options.Conventions.AddAreaFolderRoute("OrchardCore.Demo", "/Admin", _adminOptions.AdminUrlPrefix + "/Demo");
+
```

Comment:
With `IPageRouteModelConvention`  that I am currently working, it will convert 
`OrchardCore.Demo/Admin/Hello`
`OrchardCore.Demo/Account/Admin/Create`

To,
`Admin/OrchardCore.Demo/Hello`
`Admin/OrchardCore.Demo/Account/Create`

Reply:
>i think we need to fix `AddAreaFolderRoute()`
Sure, will change! . However I have one qq, why it's adding new `SelectorModel` and also keeping the original and not replacing?

Reply:
Because `AddAreaFolderRoute()` adds a route ;)

We would need a `ReplaceAreaFolderRoute()` but i don't think so, we would need to have the same behavior as for controller actions where all registered routes are still valid but the first registered routes have an higher precedence for link generation. That's why the `OC.Mvc startup Configure()` run last to register all defaults routes so that we can register routes through modules that will have an higher precedence.

So e.g, if you use an `IPageRouteModelConvention` for admin page actions, we would have to check that modules can still register routes with an higher precedence for link generation.